tests/testthat/test-labels.R

```diff
@@ -0,0 +1,116 @@
+library(magrittr)
+
+# Mock up metacore data
+var_spec <- tibble(
+   variable = names(iris),
+   length = rep(1, 5),
+   label = c("Sepal Length", "Sepal Width", "Petal Length", "Petal Width", "Species"),
+   type = c("float", "float", "float", "float", "text"),
+   format = rep(NA_character_, 5),
+   common = rep(FALSE, 5)
+)
+
+ds_spec <- tibble(
+   dataset = "Iris",
+   structure = c(""),
+   label = "Iris"
+)
+
+ds_vars <- tibble(
+   dataset = rep("Iris", 5),
+   variable = names(iris),
+   order = 1:5,
+   keep = rep(TRUE, 5),
+   key_seq = 1:5,
+   core = rep(NA_character_, 5),
+   supp_flag = rep(FALSE, 5)
+)
+
+value_spec <- tibble(
+   dataset = character(0),
+   variable = character(0),
+   origin = character(0),
+   type = character(0),
+   code_id = character(0),
+   derivation_id = character(0),
+   where = character(0)
+)
+
+derivations <- tibble(
+   derivation_id = character(0),
+   derivation = character(0)
+)
+
+code_id <- tibble(
+   code_id = character(0),
+   name = character(0),
+   type = character(0),
+   codes = list()
+   )
+
+# This is loud and I don't want it - just need the metacore object
+mc <- suppressWarnings(
+   suppressMessages(
+      metacore(ds_spec, ds_vars, var_spec, value_spec, derivations, code_id)
+      )
+   )
+
+test_that("Check that add_labels applies labels properly", {
+   x <- mtcars %>%
+      add_labels(
+         mpg = "Miles Per Gallon",
+         cyl = "Cylinders"
+      )
+
+   expect_equal(attr(x$mpg, 'label'), "Miles Per Gallon")
+   expect_equal(attr(x$cyl, 'label'), "Cylinders")
+})
+
+test_that("Check that add_labels errors properly", {
+   expect_error(add_labels(TRUE, x = "label"))
+   expect_error(add_labels(mtcars, "label"))
+   expect_error(add_labels(mtcars, bad = "label"))
+   expect_error(add_labels(mtcars, mpg = 1))
+})
+
+test_that("set_variable_labels applies labels properly", {
+
+   # Load in the metacore test object and example data
+   suppressMessages(
+      mc <- metacore::spec_to_metacore(metacore::metacore_example("p21_mock.xlsx"), quiet=TRUE) %>%
```

Comment:
I can't find where you said something about reading metacore objects from disc and the test being slow. That is why I saved a pre built metacore object and it is a bit faster that way if you are on the lastest metacore it will be there.

Reply:
Can you point me to where the saved file is?

Reply:
metacore_examples()

Reply:
I still need to change over all the test to that one so they run faster

Reply:
Ok - can I toss that one back to you so you can do it in bulk and we can just merge this one?